operators/src/processing/reprojection.rs

```diff
@@ -47,6 +47,42 @@ pub struct RasterReprojectionState {
 }
 
 pub type Reprojection = Operator<ReprojectionParams, SingleRasterOrVectorSource>;
+
+impl Reprojection {
+    fn derive_raster_result_descriptor(
+        target_sref: SpatialReference,
+        in_desc: &RasterResultDescriptor,
+    ) -> Result<RasterResultDescriptor> {
+        let out_bbox = if let Some(bbox) = in_desc.bbox {
+            let in_srs: Option<SpatialReference> = in_desc.spatial_reference.into();
+            let projector = CoordinateProjector::from_known_srs(
+                in_srs.ok_or(Error::AllSourcesMustHaveSameSpatialReference)?,
+                target_sref,
+            )?;
+
+            Some(bbox.reproject_clipped(&projector)?)
+        } else {
+            None
+        };
+
+        let out_res = match (in_desc.bbox, in_desc.resolution, out_bbox) {
+            (Some(in_bbox), Some(in_res), Some(out_bbox)) => {
+                suggest_pixel_size_from_diag_cross_projected(in_bbox, out_bbox, in_res).ok()
```

Comment:
If i remember correct we had some problems with inconsistent resolutions in the projection when using other BBOXes then the SRS ones. Is there a similar problem here? Will the resolution be the same as the one the operator actually uses? 
Also can we use the data BBOX in the operator and fall back to the SRS BBOXes?

Reply:
> If i remember correct we had some problems with inconsistent resolutions in the projection when using other BBOXes then the SRS ones.

I only remember that we had problems when we tried to partition the space and find the max resolution of all partitions. But we no longer do this.

> Will the resolution be the same as the one the operator actually uses?

What do you mean by this? The only place where we use this resolution (for now) is as a parameter in the interpolation. If the input resolution is set to `source` then this resolution here would be used to query the projection with.

> Also can we use the data BBOX in the operator and fall back to the SRS BBOXes?

I think the in_bbox and out_bbox have to be the same area, but only in different resolutions, for pixel size suggestion to work. So I guess we could compute the intersection of the in_srs_bbox and out_srs_bbox and compute the resolution on this. Shall I do it?

Now that I think about it: don't we have to do this for the case where a data bbox is available, too? (clipping the in_bbox to the area of out_bbox)

Reply:
I did some brainstorming with myself and implemented three things now

- in the query method, use the data bounds from the source result descriptor for calculating the input resolution. Only if it is not given fall back to the srs area of use
- in the derivation of the output result descriptor, I fall back to the srs area of use if the input descriptor has no bbox defined but does have a resolution defined
- in both methods, I reproject the output bbox back to the input srs now. My thinking was: we want the bbox in both projections and they should cover the same area of the world. If we only `reproject_clipped` then it might happen that the output bbox is much smaller than the input box. However, I noticed that the computed output resolution for my test drifted farther apart from what gdal does. So maybe we shouldn't do this after all? Maybe we have to TALK about it.

Reply:
yes this is also what i was thinking about. The deviation to GDAL is something that we need to talk about. GDAL also allows to process data outside of the "valid" bounds. Since we now have data bounds we could try to also allow that... However, this is something for another day.

Reply:
As we discussed I changed the computation of the corresponding bboxes for the pixel size estimation. It now takes the intersection on both boxes in WGS84. It produces exactly the same, at least in the test case.